PracticeGit/test.c

```diff
@@ -0,0 +1,129 @@
+   int numTestCases = sizeof (testCases) / sizeof (testCases[0]);
+   printf ("\n                   EXECUTE INTEGER REVERSAL TEST CASES\n"
+      "\n+-------------------+---------------------------+------------------------+----------+\n"
+      CYAN"| Original Integer  | Reversed Integer          | Palindrome Check       |  Result  |\n"RESET
+      "+-------------------+---------------------------+------------------------+----------+\n");
+   for (int i = 0; i < numTestCases; i++) {
+      int reversed, result = ReverseInteger (testCases[i], &reversed);
+      char reversedStr[12]; // Buffer for the reversed integer string
+      snprintf (reversedStr, sizeof (reversedStr), "%d", reversed);
+      if (result == SUCCESS) printf ("| %-17d | %-25d | %-22s | %-20s|\n", testCases[i], reversed,
+         IsPalindrome (reversedStr) == PALINDROME ? "Palindrome" : "Not a Palindrome", reversed == expectedResults[i] ? GREEN "Pass" RESET : RED "Fail" RESET);
+      else printf ("| %-17d | %-25s | %-22s | %-20s|\n", testCases[i], "Overflow", "Invalid input", reversed == expectedResults[i] ? GREEN "Pass" RESET : RED "Fail" RESET);
+      printf ("+-------------------+---------------------------+------------------------+----------+\n");
+   }
+}
+
+void GetPalindromeInput () {
+   char str[MAXLENGTH];
+   printf ("\nEnter a phrase or text: ");
+   if (fgets (str, MAXLENGTH, stdin) == NULL || (str[0] == '\n')) {
+      printf ("Empty input is not a valid string.\n");
+      return;
+   }
+   printf (IsPalindrome (str) == PALINDROME ? GREEN "Palindrome\n"RESET : RED"Not a palindrome\n"RESET);
+}
+
+void GetIntegerInput () {
+   char input[MAXLENGTH];
+   printf ("\nEnter an integer to reverse: ");
+   if (fgets (input, MAXLENGTH, stdin) == NULL || (input[0] == '\n')) {
+      printf ("Empty input is not a valid number.\n");
+      return;
+   }
+   char* endPtr;
+   long int number = strtol (input, &endPtr, 10);
+   if (endPtr == input || (*endPtr != '\n' && *endPtr != '\0')) {
+      printf ("Invalid input.\n");
+      return;
+   }
+   int reversed;
+   if (ReverseInteger ((int)number, &reversed) != SUCCESS) {
+      printf ("Reversed Integer: Overflow\nInvalid input\n");
+      return;
+   }
+   if (number < 0) printf ("Reversed Integer: %d\n" RED "Not a palindrome\n" RESET, reversed);
+   else printf ("Reversed Integer: %d\n%s", reversed, IsPalindrome (input) == PALINDROME ? GREEN "Palindrome\n" RESET : RED "Not a palindrome\n" RESET);
+}
+
+int main () {
+   char choiceStr[3]; // Buffer for user input
+   while (1) {
+      printf ("\nSelect option:\n1. Execute test cases\n2. Enter a phrase or text\n3. Enter an integer to reverse\n4. Exit\nEnter your choice (1, 2, 3, or 4): ");
+      fgets (choiceStr, sizeof (choiceStr), stdin);
+      char* endPtr;
+      long choice = strtol (choiceStr, &endPtr, 10);
+      if (endPtr != choiceStr && *endPtr == '\n' && choice >= 1 && choice <= 4) {

```

Comment:
By this way you can get rid of else block.
```suggestion
      if (endPtr == choiceStr || *endPtr != '\n' || choice < 1 || choice > 4) {
   printf ("Invalid choice. Please enter 1, 2, 3, or 4.\n");
   while (getchar () != '\n');  // Discard any leftover characters
   continue;
}

```